### PR TITLE
Refactor proxy config

### PR DESCRIPTION
**Pull request by simonpasquier**
Refactoring a bit the code to have the same behavior for managing infrastructure and proxy configurations. Especially CMO now remembers the last know proxy configuration so it can't get the latest version from the API, it will default to the last known version. This makes the operator more resilient to transient blips from the API server.

This change also avoids putting the operator into a Degraded status if it can't retrieve infrastructure or proxy configurations.

* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: pkg/operator/operator.go [hunk 1]
```diff
@@ -424,20 +489,47 @@ func (o *Operator) sync(key string) error {
...
+}
+
+func (o *Operator) loadProxyConfig() (*ProxyConfig, error) {
+	var proxyConfig *ProxyConfig
+
+	proxy, err := o.client.GetProxy(clusterResourceName)
+	if err != nil {
+		klog.Warningf("Error getting cluster proxy configuration: %v", err)
+
+		if proxyConfig == nil {
```
Comment (dgrisonnet) [inline pkg/operator/operator.go:522]: this should be `o.lastKnowProxyConfig`

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -424,20 +489,47 @@ func (o *Operator) sync(key string) error {
...
+	infrastructure, err := o.client.GetInfrastructure(clusterResourceName)
 	if err != nil {
-		err = errors.Wrap(err, "error getting cluster infrastructure")
-		klog.Info(err)
-		reportErr := o.client.StatusReporter().SetFailed(err, "FailedInfrastructureConfig")
-		if reportErr != nil {
-			klog.Errorf("error occurred while setting status to failed: %v", reportErr)
+		klog.Warningf("Error getting cluster infrastructure: %v", err)
+
+		if infrastructureConfig == nil {
```
Comment (dgrisonnet) [inline pkg/operator/operator.go:499]: this should be `o.lastKnowInfrastructureConfig`

File: pkg/operator/operator.go (hunk 1 above)
Comment (simonpasquier) [reply pkg/operator/operator.go]: thanks!

**Pull request merged**